### ISSUE TITLE
Could not find com.android.tools:common:25.3.1.

### ISSUE THREAD

**Issue by solonifer** [rt2zz/react-native-contacts#337]
**Old version of gradle has missing dependencies**

```
Could not find com.android.tools:common:25.3.1.
Searched in the following locations:
    https://jcenter.bintray.com/com/android/tools/common/25.3.1/common-25.3.1.pom
    https://jcenter.bintray.com/com/android/tools/common/25.3.1/common-25.3.1.jar
Required by:
    project :react-native-contacts > com.android.tools.build:gradle:2.3.1 > com.android.tools.build:gradle-core:2.3.1 > com.android.tools.build:builder:2.3.1
    project :react-native-contacts > com.android.tools.build:gradle:2.3.1 > com.android.tools.build:gradle-core:2.3.1 > com.android.tools.build:builder:2.3.1 > com.android.tools.build:manifest-merger:25.3.1
    project :react-native-contacts > com.android.tools.build:gradle:2.3.1 > com.android.tools.build:gradle-core:2.3.1 > com.android.tools.build:builder:2.3.1 > com.android.tools.ddms:ddmlib:25.3.1
    project :react-native-contacts > com.android.tools.build:gradle:2.3.1 > com.android.tools.build:gradle-core:2.3.1 > com.android.tools.build:builder:2.3.1 > com.android.tools.analytics-library:shared:25.3.1
    project :react-native-contacts > com.android.tools.build:gradle:2.3.1 > com.android.tools.build:gradle-core:2.3.1 > com.android.tools.build:builder:2.3.1 > com.android.tools.analytics-library:tracker:25.3.1
    project :react-native-contacts > com.android.tools.build:gradle:2.3.1 > com.android.tools.build:gradle-core:2.3.1 > com.android.tools.build:builder:2.3.1 > com.android.tools:sdklib:25.3.1 > com.android.tools.layoutlib:layoutlib-api:25.3.1
    project :react-native-contacts > com.android.tools.build:gradle:2.3.1 > com.android.tools.build:gradle-core:2.3.1 > com.android.tools.build:builder:2.3.1 > com.android.tools:sdklib:25.3.1 > com.android.tools:dvlib:25.3.1
    project :react-native-contacts > com.android.tools.build:gradle:2.3.1 > com.android.tools.build:gradle-core:2.3.1 > com.android.tools.build:builder:2.3.1 > com.android.tools:sdklib:25.3.1 > com.android.tools:repository:25.3.1
```

**Fix:**

```
buildscript {
    repositories {
        google()
        jcenter()
    }

    dependencies {
        classpath 'com.android.tools.build:gradle:3.1.4'
    }
}
```

**Comment by PersistentSatyenVelankar** [comment]
@solonifer  :- I encountered the same error and followed the steps mentioned above. But the error does not go away. The configuration that I am using is as follows:-

react-native-cli: 2.0.1
react-native: 0.47.0
"react": "16.0.0-alpha.12",
"react-dom": "16.0.0-alpha.12",

Request you to kindly help

**Comment by eladgel** [comment]
my solution
https://stackoverflow.com/questions/53691457/react-native-android-could-not-find-com-android-toolscommon25-2-3

**Comment by jakubtomas108** [comment]
Any working solution?

**Comment by solonifer** [comment]
@PersistentSatyenVelankar have you added google() to repositories?

**Comment by PersistentSatyenVelankar** [comment]
@solonifer - Yes I have added google() to repositories

**Comment by solonifer** [comment]
@PersistentSatyenVelankar have you changed this inside
node_modules/react-native-contacts/android/build.gradle ?
Not in your application. It should work.

You can also try to change buildToolsVersion in the same file.

**Comment by igorkurochka** [comment]
@rt2zz do you have any plans to release the fix as a new version in npmjs soon?

**Comment by PersistentSatyenVelankar** [comment]
@solonifer :- Hi , I think that means that everytime I reinstall node-modules , I need to apply this patch of code. I was looking for a solution where patching of code is not required

**Comment by morenoh149** [comment]
A PR for the fix would be merged. @PersistentSatyenVelankar @igorkurochka if someone wants to narrow down the fix to exactly the lines that need to change I'd appreciate it.

**Comment by morenoh149** [comment]
PR merged and 2.2.5 published on npm
